specification/dsdl/grammar.parsimonious

```diff
@@ -0,0 +1,178 @@
+definition  = line (end_of_line line)*  # An empty file is a valid definition. Trailing end-of-line is optional.
+line        = statement? _? comment?    # An empty line is a valid line.
+comment     = ~r"#[^\r\n]*"
+end_of_line = ~r"\r?\n"                 # Unix/Windows
+_           = ~r"[ \t]+"                # Whitespace
+
+# ==================================================== Statements ====================================================
+
+statement = statement_directive
+          / statement_service_response_marker
+          / statement_attribute
+
+statement_attribute = statement_constant
+                    / statement_field
+                    / statement_padding_field
+
+statement_constant      = type _ identifier _? "=" _? expression
+statement_field         = type _ identifier
+statement_padding_field = type_void ""  # The trailing empty symbol is to prevent the node from being optimized away.
+
+statement_service_response_marker = "---"  # Separates request/response, specifies that the definition is a service.
```

Comment:
I'd make this "3 or more" dashes. Style may dictate 3 but I don't see a benefit in being pedantic in the grammar about this.